pkgs/development/libraries/science/networking/ns-3/default.nix

```diff
@@ -1,76 +1,104 @@
 , build_profile ? null
 
-# --enable-examples
+  # --enable-examples
 , withExamples ? false
 
-# very long
-, withManual ? false, doxygen ? null, graphviz ? null, imagemagick ? null
-# for manual, tetex is used to get the eps2pdf binary
-# texlive to get latexmk. building manual still fails though
-, dia, tetex ? null, ghostscript ? null, texlive ? null
-
-# generates python bindings
-, pythonSupport ? false, ncurses ? null
-
-# All modules can be enabled by choosing 'all_modules'.
-# we include here the DCE mandatory ones
+  # very long
+, withManual ? false
+, doxygen ? null
+, graphviz ? null
+, imagemagick ? null
+  # for manual, tetex is used to get the eps2pdf binary
+  # texlive to get latexmk. building manual still fails though
+, dia
+, tetex ? null
+, ghostscript ? null
+, texlive ? null
+
+  # generates python bindings
+, pythonSupport ? true
+, ncurses ? null
+
+  # All modules can be enabled by choosing 'all_modules'.
+  # we include here the DCE mandatory ones
 , modules ? [ "core" "network" "internet" "point-to-point" "point-to-point-layout" "fd-net-device" "netanim" ]
 , lib
 }:
 
 let
-  pythonEnv = python.withPackages(ps:
+  pythonEnv = python.withPackages (ps:
     lib.optional withManual ps.sphinx
     ++ lib.optionals pythonSupport (with ps;[ pybindgen pygccxml ])
   );
 in
 stdenv.mkDerivation rec {
   pname = "ns-3";
-  version = "35";
+  version = "38";
 
   src = fetchFromGitLab {
     owner = "nsnam";
-    repo   = "ns-3-dev";
-    rev    = "ns-3.${version}";
-    sha256 = "sha256-3w+lCWWra9sndL8+vkGfH5plrDYYCMFi1PzwIVRku6I=";
+    repo = "ns-3-dev";
+    rev = "ns-3.${version}";
+    sha256 = "sha256-3t1Ghdp2jf2op567+YpBP/EGDZjg/tAc7+QNUDnfFPg=";
   };
 
-  nativeBuildInputs = [ wafHook python ];
+  nativeBuildInputs = [ cmake pkg-config pythonEnv autoPatchelfHook ];
 
-  outputs = [ "out" ] ++ lib.optional pythonSupport "py";
+  outputs = [ "out" ];
 
   # ncurses is a hidden dependency of waf when checking python
   buildInputs = lib.optionals pythonSupport [ castxml ncurses ]
     ++ lib.optionals enableDoxygen [ doxygen graphviz imagemagick ]
-    ++ lib.optionals withManual [ dia tetex ghostscript texlive.combined.scheme-medium ];
+    ++ lib.optionals withManual [ dia tetex ghostscript imagemagick texlive.combined.scheme-medium ]
+    ++ [
+    libxml2
+    pythonEnv
+    sqlite.dev
+    dpdk
+    openssl.dev
+    gsl
+    boost
+    root
+    glib.out
+    glib.dev
+    libpcap
+    libnl
+    libelf
+    jansson
+    pcre2.dev
+  ];
 
   propagatedBuildInputs = [ pythonEnv ];
 
-  postPatch = ''
-    patchShebangs doc/ns3_html_theme/get_version.sh
-  '';
-
-  wafConfigureFlags = with lib; [
-      "--enable-modules=${concatStringsSep "," modules}"
-      "--with-python=${pythonEnv.interpreter}"
-  ]
-  ++ optional (build_profile != null) "--build-profile=${build_profile}"
-  ++ optional withExamples " --enable-examples "
-  ++ optional doCheck " --enable-tests "
```

Comment:
Are ns-3 tests still executed in case doCheck is at true, since the --enable-tests part seems absent ?

Reply:
@rgrunbla I checked and they don't! However the build hangs or takes forever when enabled. So I disabled them.